lib/src/main/java/graphql/nadel/engine/transform/hydration/NadelHydrationTransform.kt

```diff
@@ -75,18 +82,24 @@ internal class NadelHydrationTransform(
         overallField: ExecutableNormalizedField,
         hydrationDetails: ServiceExecutionHydrationDetails?,
     ): State? {
-        val hydrationInstructionsByTypeNames = executionBlueprint.fieldInstructions
+        val instructionsByObjectTypeName = executionBlueprint.fieldInstructions
             .getTypeNameToInstructionsMap<NadelHydrationFieldInstruction>(overallField)
+            .takeIf {
+                it.isNotEmpty()
+            }
+            ?: executionBlueprint.getInstructionInsideVirtualType<NadelHydrationFieldInstruction>(executionBlueprint, hydrationDetails, overallField)
```

Comment:
So this new code will execute for every field that is not mapped to a regular hydration instruction - that's a lot of potential fields to trigger the new code. Blueprint errors are much safer, since they'll blow up at schema parsing/creating time, but potential runtime errors scare me a bit. 
Have you considered putting the new execution code behind a hint?